source/common/http/alternate_protocols_cache_impl.h

```diff
@@ -100,8 +106,34 @@ class AlternateProtocolsCacheImpl : public AlternateProtocolsCache,
     }
   };
 
+  using ProtocolsMap = quiche::QuicheLinkedHashMap<Origin, OriginData, OriginHash>;
   // Map from origin to list of alternate protocols.
-  quiche::QuicheLinkedHashMap<Origin, OriginData, OriginHash> protocols_;
+  ProtocolsMap protocols_;
+
+  // This allows calling setAlternativesImpl without creating an additional copy
+  // of the protocols vector.
+  struct OriginDataWithOptRef {
+    OriginDataWithOptRef()
+        : protocols{}, srtt(std::chrono::milliseconds(0)), h3_status_tracker(nullptr) {}
+    OriginDataWithOptRef(OptRef<std::vector<AlternateProtocol>> p, std::chrono::microseconds s,
+                         Http3StatusTrackerPtr&& t)
```

Comment:
nit: here and elsewhere, it's totally fine to have the name of the parameter match the name of the member. So these paramaters can be named protocols, srtt and h3_status_tracker. Then we just have the initializer list:
`protocols(protocols), srtt(srtt) and h3_status_tracker(h3_status_tracker)`
This avoid the need for less readable names.

Reply:
done